### ISSUE TITLE
Logical OR (`||`) in TPLs is recognized as "invalid modifier"

### ISSUE THREAD

**Issue by jakub-g** [ariatemplates/ariatemplates#769]
Consider the following TPL snippet:

http://instant.ariatemplates.com/anonymous/0cd69ce1cf6f8918e2a7/1f0bd7

```
{macro main()}

  <h1>1: ${ (true) && (true) }</h1>
  <h1>2: ${ (true) && (false) }</h1>
  <h1>3: ${ (5 + 12*12) % 100 }</h1>
  <h1>4: ${ (5==5) ? "a" : "b"}</h1>

  //template error: invalid modifier syntax
  //<h1>5: ${ 8 | 16 }</h1>
  //<h1>6: ${ (null) || ("blah") }</h1>

{/macro}
```

All of the valid javascript expressions inside `${}` work fine in TPL except the expressions having `|` in it, i.e. the single pipe `|` for bitwise OR and double pipe for logical or `||`.

This is clearly not so nice, since the user might expect that all legal expressions inside of `${}` will work fine.

For instance the typical JS construct like `${some.variable||"default value"}` won't work (though we have a related modifier, `${some.variable|empty:"default value"}` which gets activated on any falsy value).

I see three possible solutions to it:
1. Don't care about single pipe `|` being bitwise OR, and treat it still as a modifier, but when two pipes are in place i.e. `||`, then let this mean logical OR.
   In other words, tweak the `|` processing in the TPLs as a modifiers so that double pipe `||` is not tried being turned into a modifier.
2. (radical) Change the special char for modifiers to sth that's not a JS special char. E.g. `#`.
3. Live with it and close this issue.

What does the team think?
